lib/index.js

```diff
@@ -682,6 +683,13 @@ var Client = module.exports = function(config) {
                     basic = new Buffer(this.auth.username + ":" + this.auth.password, "ascii").toString("base64");
                     headers["Authorization"] = "Basic " + basic;
                     break;
+                case "netrc":
+                    var auth = netrc()["api.github.com"];
```

Comment:
In that particular way this wouldn't work, as you specifically must have the `api.github.com` domain listed there. However, I'm pretty sure it will share the same password as `api.github.com`, so there would be no need to add `uploads.github.com` there. It will just work as it is.